python/triton/language/semantic.py

```diff
@@ -1322,6 +1322,18 @@ def _str_to_dot_input_precision(input_precision, builder):
 def dot(lhs: tl.tensor, rhs: tl.tensor, acc: tl.tensor, input_precision: Optional[str], max_num_imprecise_acc: int,
         out_dtype: tl.dtype, builder: ir.builder) -> tl.tensor:
 
+    # Test for supporting reduction axis `k == 8`
+    def support_m16n16k8():
+        if not hasattr(builder.options, "arch"):
+            return False
+        archStr = str(builder.options.arch)
+        # CDNA 3.0 supports k==8 in all mfma variants except for int8
+        # (where the smallest `k` supported is 16)
+        if "gfx94" in archStr:
```

Comment:
I wonder if we could have a min_dot_size parameter in the backend instead of explicitly checking for the target name? That would help other downstream backend (https://github.com/triton-lang/triton/pull/4271)
Is this something you considered?

Reply:
Hi @ThomasRaoux , the problem I see with that `min_dot_size` should depend on the data type. For Nvidia this would be (as far I can see) >= 16 for float and >= 32 for int. For us, we would have gfx94x would support k >= 8 for float and k>=16 for int. Maybe  a function? 

But if a function, why not having a function inside the dot op (maybe a bit more general of what I wrote)?

Reply:
Good point, then yes maybe having a lambda set by the backend is the way to go?

Reply:
Just to clarify, you mean having a function in `{HIP,Nvidia}Options` right?

Reply:
But in `semantic.py` we don't have access to any `BaseBackend` derived class (I think), this is why I was mentioning the options. We can get the options from the `ir.builder`, but I am not sure we can get the backend from the `ir.builder`. Is there a way I can access the backend from `semantic.py`?

Reply:
Ah ok, I see. Those `codegen_fn` are attached to the `builder`. So we can use that in `semantic.py`. I will try this out

Reply:
I had a go at it. I (bravely) changed also nvidia stuff, surely breaking tests. But I wanted to agree on the design before fixing those.

Reply:
Ha interesting. this is certainly better! :)

Reply:
Thanks for doing that, it looks good!